docs/csharp/language-reference/keywords/snippets/PassParameters.cs

```diff
@@ -1,47 +1,235 @@
 ﻿using System;
 
-namespace Keywords;
+namespace MethodParameters;
 
-// <PassByValueOrReference>
-class TheClass
+// <DataTypes>
+public record struct Point(int X, int Y);
+public record class Point3D
```

Comment:
Since this is a `record`, I'd expect primary constructor semantics here. The only difference you're trying to call attention to is `class` vs `struct`, right?

I'd make this `public record class Point3D(int X, int Y, int Z);`

Unless, you're trying to assign property values later on like `point3D.X = 7;` and not use `with`.

Reply:
It's intentional. `with`ers are *non-destructive* mutation. That make a new instance, rather than changing the current instance. I'll add a comment to make that clear.

```suggestion
// This doesn't use a primary constructor because the properties implemented for `record` types are 
// readonly in record class types. That would prevent the mutations necessary for this example.
public record class Point3D
```